File: lib/src/main/java/com/darklycoder/wifitool/lib/interfaces/WiFiStatusListener.java
package com.darklycoder.wifitool.lib.interfaces;

import android.content.Intent;

/**
 * WiFi状态回调
 */
public interface WiFiStatusListener {

    /**
     * 处理扫描列表
     */
    void handleScanResultsChanged(Intent intent);

    /**
     * 处理WiFi状态
     */
    void handleWiFiStateChanged(Intent intent);

    /**
     * 处理网络状态
     */
    void handleNetStateChanged(Intent intent);

    /**
     * 处理WiFi状态(密码错误回调)
     */
    void handleSupplicantStateChanged(Intent intent);

}

File: lib/src/main/java/com/darklycoder/wifitool/lib/interfaces/WiFiStatusListener.kt
package com.darklycoder.wifitool.lib.interfaces

import android.content.Intent

/**
 * WiFi状态回调
 */
interface WiFiStatusListener {

    /**
     * 处理扫描列表
     */
    fun handleScanResultsChanged(intent: Intent)

    /**
     * 处理WiFi状态
     */
    fun handleWiFiStateChanged(intent: Intent)

    /**
     * 处理网络状态
     */
    fun handleNetStateChanged(intent: Intent)

    /**
     * 处理WiFi状态(密码错误回调)
     */
    fun handleSupplicantStateChanged(intent: Intent)

}
